File: frontend/src/api/flights.ts
import axios from "axios";

const API_BASE_URL = "http://127.0.0.1:8000";

export interface Flight {
    id: number;
    origin: string;
    destination: string;
    status: string;
}

export interface TrajectoryPoint {
    lat: number;
    lon: number;
}

export const createFlight = async (origin: string, destination: string) => {
    const response = await axios.post<Flight>(`${API_BASE_URL}/flights/`, {
        origin,
        destination,
    });
    return response.data;
};

export const getFlightTrajectory = async (flightId: number) => {
    const response = await axios.get(
        `${API_BASE_URL}/flights/${flightId}/trajectory`
    );
    return response.data;
};

File: backend/app/routes/flights.py
# app/routes/flights.py
from fastapi import APIRouter
from app.models.flight import FlightCreate, Flight
from ai.trajectory import generate_ai_trajectory

router = APIRouter(prefix="/flights", tags=["Flights"])

flights_db = {}
flight_counter = 1


@router.post("/")
def create_flight(flight: FlightCreate):
    global flight_counter
    
    trajectory = generate_ai_trajectory(
        flight.origin,
        flight.destination
    )

    new_flight = Flight(
        id=flight_counter,
        origin=flight.origin,
        destination=flight.destination,
        status="planned",
        trajectory=trajectory
    )

    flights_db[flight_counter] = new_flight
    flight_counter += 1

    return new_flight


@router.get("/{flight_id}/trajectory")
def get_flight_trajectory(flight_id: int):
    flight = flights_db.get(flight_id)

    if not flight:
        return {"error": "Flight not found"}

    return {
        "flight_id": flight.id,
        "origin": flight.origin,
        "destination": flight.destination,
        "trajectory": flight.trajectory
    }
